time_execution/backends/threaded.py

```diff
@@ -24,6 +24,7 @@ def __init__(self, backend, backend_args=None, backend_kwargs=None,
             backend_args = tuple()
         if backend_kwargs is None:
             backend_kwargs = dict()
+        self.parent_thread = threading.current_thread()
```

Comment:
I'm assuming that when the calling thread ends the timeexecution thread should stop as well